Sources/StreamChat/Utils/Codable+Extensions.swift

```diff
@@ -37,6 +51,9 @@ extension JSONDecoder {
             throw DecodingError.dataCorruptedError(in: container, debugDescription: "Invalid date: \(dateString)")
         }
         
+        // We enable the UserPayload caching by setting this key
+        decoder.userInfo[userPayloadCachingFlagKey] = true
```

Comment:
Why is there this flag in `userInfo` while it is always true and integrators cannot change it? I would skip it and treat it as always on

Reply:
Or is there any reason why we want to enable/disable it?

Reply:
We want the `UserPayload` to be decodable with any JSONDecoder, not just with ours. Without the flag/subclass, cache will always fill, in turn we'll get invalid cache across decodings. Without this flag, decoding `UserPayload` with other JSONDecoder's would fill the cache, and would result in incorrect decodings

Reply:
UserPayload is decodable with any JSONDecoder. And with the current logic, if the value is not in the cache, it will decode it from the json. I don't see a problem here. Am I missing anything?

Reply:
Without the flag, using any other JSONDecoder:
1. You decode a ChannelListPayload. All UserPayloads are cached.
2. You decode another ChannelListPayload, with the same users, but they're updated. All UserPayloads are received from the cache (because their id is not changed), but they're outdated